sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala

```diff
@@ -217,6 +218,24 @@ object SchemaUtil {
     }
   }
 
+  def checkMissingCols(colsLeft: Seq[String], colsRight: Seq[String], caseSensitive: Boolean): Seq[String] = {
+    if (caseSensitive) colsLeft.diff(colsRight)
+    else colsLeft.map(_.toLowerCase).diff(colsRight.map(_.toLowerCase))
+  }
+
+  def checkPartitionMatch(configuredPartitions: Seq[String], existingPartitions: Seq[String], caseSensitive: Boolean): (Boolean, Set[String], Set[String]) = {
+    val (confPartitions, existPartitions) = if (caseSensitive) {
+      val conf = configuredPartitions.toSet
+      val existing = existingPartitions.toSet
+      (conf, existing)
+    } else {
```

Comment:
Why not a little bit shorter?
```suggestion
    val (confPartitions, existPartitions) = if (caseSensitive) (configuredPartitions.toSet, existingPartitions.toSet)
    else {
```